pkg/vault/contracts/AddUnbalancedLiquidityViaSwapRouter.sol

```diff
@@ -138,75 +139,69 @@ contract AddUnbalancedLiquidityViaSwapRouter is RouterQueries, IAddUnbalancedLiq
     function _computeAddUnbalancedLiquidityViaSwap(
         AddLiquidityAndSwapHookParams calldata hookParams
     ) private returns (IERC20[] memory tokens, uint256[] memory amountsIn, uint256[] memory amountsOut) {
+        // maxAmountsIn length is checked against tokens length at the Vault.
+        tokens = _vault.getPoolTokens(hookParams.pool);
+        amountsOut = new uint256[](tokens.length);
+        uint256[] memory maxAmountsIn = new uint256[](tokens.length);
+
+        uint256 exactTokenIndex = hookParams.operationParams.exactToken == tokens[0] ? 0 : 1;
+        uint256 adjustableTokenIndex = exactTokenIndex == 0 ? 1 : 0;
+
+        maxAmountsIn[exactTokenIndex] = hookParams.operationParams.maxAdjustableAmount;
+        maxAmountsIn[adjustableTokenIndex] = hookParams.operationParams.maxAdjustableAmount;
```

Comment:
As discussed off-thread, I _think_ we should just use _MAX_AMOUNT as the limit for this first operation, and check limits later.